src/metabase/driver/sql_jdbc/sync/describe_database.clj

```diff
@@ -164,15 +196,11 @@
     db-or-id-or-spec
     nil
     (fn [^Connection conn]
-      (let [schema-filter-prop      (driver.u/find-schema-filters-prop driver)
-            has-schema-filter-prop? (some? schema-filter-prop)
-            default-active-tbl-fn   #(into #{} (sql-jdbc.sync.interface/active-tables driver conn nil nil))]
-        (if has-schema-filter-prop?
-          (if-let [database (db-or-id-or-spec->database db-or-id-or-spec)]
-            (let [prop-nm                                 (:name schema-filter-prop)
-                  [inclusion-patterns exclusion-patterns] (driver.s/db-details->schema-filter-patterns
-                                                           prop-nm
-                                                           database)]
-              (into #{} (sql-jdbc.sync.interface/active-tables driver conn inclusion-patterns exclusion-patterns)))
-            (default-active-tbl-fn))
-          (default-active-tbl-fn)))))})
+      (let [schema-filter-prop (driver.u/find-schema-filters-prop driver)
+            database           (db-or-id-or-spec->database db-or-id-or-spec)]
+        (if (some? schema-filter-prop)
+          (let [prop-nm              (:name schema-filter-prop)
```

Comment:
I've removed the branch were database is nil. doesn't make any sense to call `describe-database` on a database does not exist.

Reply:
I think it was designed so that it can takes a spec as argument. I don't see anywhere that's used in production, only in tests.

but to be safe I will revert this change and follow up on this later. don't want another PR to have unnesscary change then turns out it's nessceary like #37620 :)